test/connection.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-env node, mocha */
+require('./init.js');
+var assert = require('assert');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+
+var config;
+
+before(function() {
+  config = global.getConfig();
+  console.log("config", config);
+});
+
+describe('testConnection', function() {
+  it('should pass with valid settings', function(done) {
+    var db = new DataSource(require('../'), config);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
```

Comment:
This assert is redundant and will make troubleshooting difficult, because it hides error details. The right way for handling errors is to call `if (err) done(err)`.

In this case, I think it's ok to simply pass `done` as the callback for `ping`:

```js
// the test passes when the connection can be established
db.ping(done);
```